Examples/Io/HepMC3/include/ActsExamples/Io/HepMC3/HepMC3Writer.hpp

```diff
@@ -1,33 +1,51 @@
 // This file is part of the Acts project.
 //
-// Copyright (C) 2018 CERN for the benefit of the Acts project
+// Copyright (C) 2018-2020 CERN for the benefit of the Acts project
 //
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 #pragma once
 
-#include <HepMC3/GenEvent.h>
+#include "ActsExamples/Framework/WriterT.hpp"
+#include "ActsExamples/Utilities/OptionsFwd.hpp"
+#include <string>
+#include <HepMC/GenEvent.h>
 #include <HepMC3/WriterAscii.h>
 
 namespace ActsExamples {
 
 /// HepMC3 event writer.
-struct HepMC3WriterAscii {
- public:
-  /// @brief Writes an event to file
-  /// @param writer writer of run files
-  /// @param event storage of the event
-  /// @return boolean indicator if the writing was successful
-  /// @note HepMC3 does not state a success or failure. The returned argument is
-  /// always true.
-  bool writeEvent(HepMC3::WriterAscii& writer,
-                  std::shared_ptr<HepMC3::GenEvent> event);
+struct HepMC3WriterAscii final
+    : public WriterT<std::vector<std::shared_ptr<HepMC3::GenEvent>>> {
+  struct Config {
+    // The output directory
+    std::string outputDir = "";
+    // The stem of output file names
+    std::string outputStemFileName = "";
```

Comment:
```suggestion
    std::string outputStem;
```
Stem is already sufficient. This is also consistent with the other configurations, e.g. the csv reader/writer.

Reply:
done